src/expressionDescriptor.ts

```diff
@@ -65,10 +74,44 @@ export class ExpressionDescriptor {
     localesLoader.load(ExpressionDescriptor.locales);
   }
 
+  static toDetails(
+    expression: string,
+    {
+      throwExceptionOnParseError = true,
+      verbose = false,
+      dayOfWeekStartIndexZero = true,
+      monthStartIndexZero = false,
+      use24HourTimeFormat,
+      locale = null,
+    }: Options = {}
+  ) {
+    // We take advantage of Destructuring Object Parameters (and defaults) in TS/ES6 and now we will reassemble back to
+    // an Options type so we can pass around options with ease.
+
+    let options = <Options>{
+      throwExceptionOnParseError: throwExceptionOnParseError,
+      verbose: verbose,
+      dayOfWeekStartIndexZero: dayOfWeekStartIndexZero,
+      monthStartIndexZero: monthStartIndexZero,
+      use24HourTimeFormat: use24HourTimeFormat,
+      locale: locale,
+    };
+
+    let descripter = new ExpressionDescriptor(expression, options);
+    descripter.parseDetails.full = descripter.getFullDescription();
+    descripter.getTimeOfDayDescription();
+    descripter.parseDetails.mond = descripter.getDayOfMonthDescription();
+    descripter.parseDetails.mon = descripter.getMonthDescription();
+    descripter.parseDetails.wkd = descripter.getDayOfWeekDescription();
+    descripter.parseDetails.year = descripter.getYearDescription();
```

Comment:
In your example you show:

```
wkd: ", only on Tuesday"
year: ", only in 2022"
```

Can we remove proceeding comma and spacing to clean it up?

Reply:
it's ok if needed